### ISSUE TITLE
Uncaught TypeError: Cannot read property 'then' of undefined

### ISSUE THREAD

**Issue by petercr** [atom/atom#15661]
[Enter steps to reproduce:]

1. Removed `ide-typescript` from packages becuase it would not update, and I keep getting the error `EBUSY: resource busy or locked, rmdir 'C:\Users\Cruckshank\.atom\packages\ide-typescript'`
2.  I re-installed `ide-typescript` to see if this would fix the problem.
3. I hit the **Update All** button and received the error
![2017-09-15 1](https://user-images.githubusercontent.com/26460352/30489428-30733684-9a05-11e7-8e4b-0ca46ff54da1.png)

**Atom**: 1.21.0-beta0 x64
**Electron**: 1.6.9
**OS**: Microsoft Windows 10 Home
**Thrown From**: Atom Core

### Stack Trace

Uncaught TypeError: Cannot read property 'then' of undefined

```
At C:\Users\Cruckshank\AppData\Local\atom\app-1.21.0-beta0\resources\app\node_modules\settings-view\lib\updates-panel.js:312

TypeError: Cannot read property 'then' of undefined
    at UpdatesPanel.updateAll (~/AppData/Local/atom/app-1.21.0-beta0/resources/app/node_modules/settings-view/lib/updates-panel.js:312:37)
    at HTMLButtonElement.onclick (~/AppData/Local/atom/app-1.21.0-beta0/resources/app/node_modules/settings-view/lib/updates-panel.js:152:36)
```

### Commands

```
```

### Non-Core Packages

```
an-old-hope-syntax 1.0.4 
atom-bootstrap3 1.2.12 
atom-ide-ui 0.3.3 
atom-live-server 2.1.2 
busy-signal 1.4.3 
emmet 2.4.3 
file-icons 2.1.11 
ide-typescript 0.1.6 
intentions 1.1.5 
linter 2.2.0 
linter-eslint 8.2.1 
linter-ui-default 1.6.9 
minimap 4.29.6 
pigments 0.40.2 
pristine-ui 1.1.9 
pure-syntax 0.3.0 
```

**Comment by 50Wliu** [comment]
Tracking at atom/settings-view#879.

**Comment by ineil** [comment]
Still occurring when updating.

**Atom**: 1.21.0-beta2 x64
**Electron**: 1.6.9
**OS**: Mac OS X 10.13
**Thrown From**: Atom Core


### Stack Trace

Uncaught TypeError: Cannot read property 'then' of undefined

```
At /Applications/Atom.app/Contents/Resources/app/node_modules/settings-view/lib/updates-panel.js:312

TypeError: Cannot read property 'then' of undefined
    at UpdatesPanel.updateAll (/Applications/Atom.app/Contents/Resources/app/node_modules/settings-view/lib/updates-panel.js:312:37)
    at HTMLButtonElement.onclick (/Applications/Atom.app/Contents/Resources/app/node_modules/settings-view/lib/updates-panel.js:152:36)
```

### Commands

```
```

### Non-Core Packages

```
atom-ide-ui 0.4.0 
autoclose-html 0.23.0 
autocomplete-json 5.5.0 
autocomplete-modules 1.7.1 
autocomplete-webpack 0.9.0 
busy-signal 1.4.3 
d3-snippets 1.0.2 
es6-javascript 1.0.0 
file-icons 2.1.12 
git-plus 7.9.3 
git-time-machine 1.5.9 
goto 1.8.3 
highlight-selected 0.13.1 
hyperclick 0.0.0 
ide-flowtype 0.17.4 
ide-json 0.1.0 
ide-vue 0.1.7 
intentions 1.1.5 
js-hyperclick 1.11.1 
language-babel 2.74.4 
linter 2.2.0 
linter-csslint 2.0.0 
linter-eslint 8.3.2 
linter-sass-lint 1.8.0 
linter-ui-default 1.6.10 
minimap 4.29.7 
minimap-cursorline 0.2.0 
minimap-find-and-replace 4.5.2 
minimap-git-diff 4.3.1 
minimap-highlight-selected 4.6.1 
minimap-selection 4.5.0 
prettier-atom 0.40.0 
pretty-json 1.6.4 
react-es7-snippets 0.3.13 
tree-view-git-status 1.4.0 
turbo-javascript 1.4.0 
```

**Comment by LucasArruda** [comment]
Also happened to me

```
/Applications/Atom.app/Contents/Resources/app/node_modules/settings-view/lib/updates-panel.js:312
Hide Stack Trace
TypeError: Cannot read property 'then' of undefined
    at UpdatesPanel.updateAll (/Applications/Atom.app/Contents/Resources/app/node_modules/settings-view/lib/updates-panel.js:312:37)
    at HTMLButtonElement.onclick (/Applications/Atom.app/Contents/Resources/app/node_modules/settings-view/lib/updates-panel.js:152:36)
```

I clicked `Update All` and it was still showing `UPDATE` for the second package (`language-babel`). I clicked `Update All` again and the error triggered.

<img width="1131" alt="screen shot 2018-01-04 at 7 11 58 pm" src="https://user-images.githubusercontent.com/88818/34584595-8fb26928-f183-11e7-8269-10f25169c88c.png">

I don't think it's related to a specific package, but the fact a package is supposedly not updating but when you click `update all`, because the package was in fact updating, an error occurs.

**Comment by 50Wliu** [comment]
This is scheduled to be fixed in Atom 1.25.0.
